packages/office-addin-unit-test/src/officeJSMock.ts

```diff
@@ -0,0 +1,150 @@
+import { usageDataObject } from "./defaults";
+
+export class OfficeJSMock {
+  constructor(object?: OfficeObject) {
+    this.properties = new Map<string, OfficeJSMock>();
+    this.loaded = false;
+    this.resetValue(undefined);
+    if (object) {
+      this.populate(object);
+    }
+  }
+
+  // Adds a function to OfficeJSMock
+  addMockFunction(methodName: string, functionality?: Function) {
+    this[methodName] = functionality ? functionality : function () {};
+  }
+
+  // Adds a OfficeJSMock to OfficeJSMock
+  addMockObject(objectName: string) {
```

Comment:
I'm confuse at the intention of this method.  We're creating a new instance of ourself and allowing it to be done multiple times with different names?

Reply:
This method is to add a new object to our Office-js Mock.
Example:
```
range = new OfficeJSMock();
range.addMockObject("font");
```
Would create the following format:
```
{
    range: {
        font: {}
    }
}
```
Where both range and font are OfficeJSMocks.
I will change the commentary to explain better what the user needs to know about this function

Reply:
The name "OfficeJSMock" sound like the top level mock of the whole thing rather than each layer within the whole API.  That is what is confusing me.

I also think now that we don't need the term "Object" in the name . . . just 'addMock' or something like that.

Reply:
Yeah, "OfficeJSMock" is acting as both the top layer as each layer of objects. Every object is an OfficeJSMock. Do you think of any other name I could change it to to be more meaningful? I am thinking of OfficeJSMockObject if you think it would be better.

Reply:
That name sill leaves me thinking it's the top level only,  Something like OfficeJSObjectMock sounds more like we're mocking one of the objects instead of just the top layer . . . thought I'm still not excited about that name.  Right now I'm also questioning the need for "JS" in the name . . . and maybe that is what is triggering the idea of the top level object.  Maybe something like "OfficeObjectMock" or "OfficeMockObject"?  These sound better to me, but I'm still not entirely sure.  Maybe @LouMM or @akrantz might have some thoughts here too.

Reply:
I like the idea of removing JS from the name. This is all javascript, so it is already implicit that it is related to JS.
I changed the name to `OfficeMockObject` for now